src/components/TaskDetails/TaskDetails.jsx

```diff
@@ -46,94 +40,98 @@ export default class TaskDetails extends React.Component {
     });
   };
 
-  handleSaveTaskDetails =(taskName, taskDescription, taskStatus, taskPriority) => {
-    const {
-      name, description, status, priority,
-    } = this.state;
-    const { taskRef } = this.props;
-    this.setState(() => ({
-      name: taskName || name,
-      description: taskDescription || description,
-      status: taskStatus || status,
-      priority: taskPriority || priority,
-    }));
-    const { subtaskList } = this.state;
-    const task = {
-      name: taskName || name || {},
-      description: taskDescription || description || {},
-      status: taskStatus || status || {},
-      priority: taskPriority || priority || {},
-      subtaskList: subtaskList || {},
+    handleSaveTaskDetails = (taskName, taskDescription, taskStatus, taskPriority) => {
+      const { taskRef, taskDetailsActions, taskDetails } = this.props;
+      const updatedTaskDetails = {
+        name: taskName || taskDetails.name || {},
```

Comment:
If I understand right, name and description are strings.... then default value (in case if user didn't input any name or description) should not be an empty object, but an empty string, like this:
      name: taskName || name || '',
      description: taskDescription || description || '',